cmd/data-aggregation-api/main.go

```diff
@@ -56,7 +56,9 @@ func run() error {
 	// TODO: be able to close goroutine when the context is closed (graceful shutdown)
 	go job.StartBuildLoop(&deviceRepo, &reports)
 
-	router.NewManager(&deviceRepo, &reports).ListenAndServe(ctx, config.Cfg.API.ListenAddress, config.Cfg.API.ListenPort)
+	if err := router.NewManager(&deviceRepo, &reports).ListenAndServe(ctx, config.Cfg.API.ListenAddress, config.Cfg.API.ListenPort); err != nil {
+		log.Error().Err(err).Msg("webserver error")
```

Comment:
Why do not you want to make this error fatal ?

Reply:
done

Reply:
actually the defer stop() will not run if we use Fatal():

```
Lint and test: cmd/data-aggregation-api/main.go#L60
exitAfterDefer: log.Fatal will exit, and `defer stop()` will not run (gocritic)
```